packages/server/test/websockets.test.ts

```diff
@@ -60,18 +61,16 @@ function factory(config?: { createContext: () => Promise<any> }) {
       .subscription('onMessage', {
         input: z.string().nullish(),
         resolve() {
-          const sub = (subRef.current = new trpc.Subscription<Message>(
-            (emit) => {
-              const onMessage = (data: Message) => {
-                emit.data(data);
-              };
-              ee.on('server:msg', onMessage);
-              return () => {
-                subscriptionEnded();
-                ee.off('server:msg', onMessage);
-              };
-            },
-          ));
+          const sub = (subRef.current = observable<Message>((emit) => {
```

Comment:
tests in this file are failing because `subRef.current` is supposed to be the subscription`. However now that we are returning an observable, there is no direct access to the subscription, so not sure how to fix those tests.